thirdeye/thirdeye-pinot/src/test/java/com/linkedin/thirdeye/alert/feed/TestUnionAnomalyFeed.java

```diff
@@ -0,0 +1,88 @@
+package com.linkedin.thirdeye.alert.feed;
+
+import com.linkedin.thirdeye.alert.commons.AnomalyFeedConfig;
+import com.linkedin.thirdeye.api.DimensionMap;
+import com.linkedin.thirdeye.datalayer.DaoTestUtils;
+import com.linkedin.thirdeye.datalayer.bao.AlertSnapshotManager;
+import com.linkedin.thirdeye.datalayer.bao.AnomalyFunctionManager;
+import com.linkedin.thirdeye.datalayer.bao.DAOTestBase;
+import com.linkedin.thirdeye.datalayer.bao.MergedAnomalyResultManager;
+import com.linkedin.thirdeye.datalayer.dto.AlertSnapshotDTO;
+import com.linkedin.thirdeye.datalayer.dto.AnomalyFunctionDTO;
+import com.linkedin.thirdeye.datalayer.dto.MergedAnomalyResultDTO;
+import com.linkedin.thirdeye.datasource.DAORegistry;
+import com.linkedin.thirdeye.detector.email.filter.AlertFilterFactory;
+import java.util.Collection;
+import org.testng.Assert;
+import org.testng.annotations.AfterClass;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.Test;
+
+
+public class TestUnionAnomalyFeed {
+  private DAOTestBase testDAOProvider;
+  private AlertFilterFactory alertFilterFactory;
+  private AlertSnapshotManager alertSnapshotDAO;
+  private AnomalyFeedConfig anomalyFeedConfig;
+  private MergedAnomalyResultManager mergedAnomalyResultDAO;
+  private AnomalyFunctionManager anomalyFunctionDAO;
+
+  private static String TEST = "test";
+  private long alertSnapshotId;
+  @BeforeClass
+  void beforeClass() {
+    testDAOProvider = DAOTestBase.getInstance();
+    String mappingsPath = ClassLoader.getSystemResource("sample-alertfilter.properties").getPath();
+    alertFilterFactory = new AlertFilterFactory(mappingsPath);
+    DAORegistry daoRegistry = DAORegistry.getInstance();
+    alertSnapshotDAO = daoRegistry.getAlertSnapshotDAO();
+    mergedAnomalyResultDAO = daoRegistry.getMergedAnomalyResultDAO();
+    anomalyFunctionDAO = daoRegistry.getAnomalyFunctionDAO();
+    init();
+  }
+
+  @AfterClass(alwaysRun = true)
+  void afterClass() {
+    testDAOProvider.cleanup();
+  }
+
+  private void init() {
+    AlertSnapshotDTO alertSnapshotDTO = DaoTestUtils.getTestAlertSnapshot();
+    alertSnapshotId = alertSnapshotDAO.save(alertSnapshotDTO);
```

Comment:
Does this mean we have to create alertSnapshot manunally for each AnomalyFeed?

Reply:
No. The initial creation of alertSnapshot is handled in AlertTaskRunnerV2.